neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/dialect/impl/EntityQueries.java

```diff
@@ -0,0 +1,667 @@
+	}
+
+	private Map<String, String> initRemovePropertyQueries(EntityKeyMetadata entityKeyMetadata, TupleContext tupleContext) {
+		if ( tupleContext == null ) {
+			return Collections.emptyMap();
+		}
+
+		Map<String, String> removeColumn = new HashMap<>();
+		for ( String column : tupleContext.getSelectableColumns() ) {
+			if ( !column.contains( "." ) ) {
+				StringBuilder queryBuilder = new StringBuilder();
+				queryBuilder.append( "MATCH " );
+				appendEntityNode( "n", entityKeyMetadata, queryBuilder );
+				queryBuilder.append( " REMOVE n." );
+				escapeIdentifier( queryBuilder, column );
+
+				removeColumn.put( column, queryBuilder.toString() );
+			}
+		}
+		return Collections.unmodifiableMap( removeColumn );
+	}
+
+	/*
+	 * This method will initialize the query string for a multi get.
+	 * The query is different in the two scenarios:
+	 * 1) the id is mapped on a single property:
+	 *
+	 * MATCH (n:ENTITY:table)
+	 * WHERE n.id IN {0}
+	 * RETURN n
+	 *
+	 * 2) id is mapped on multiple columns:
+	 *
+	 * MATCH (n:ENTITY:table)
+	 * WHERE
+	 *
+	 * In this case the query depends on how many id we are retrieving and it will be completed later
+	 */
+	private static String initMultiGetEntitiesQuery(EntityKeyMetadata entityKeyMetadata) {
+		StringBuilder queryBuilder = new StringBuilder( "MATCH " );
+		queryBuilder.append( "(n:" );
+		queryBuilder.append( ENTITY );
+		queryBuilder.append( ":" );
+		appendLabel( entityKeyMetadata, queryBuilder );
+		queryBuilder.append( ") ");
+		queryBuilder.append( " WHERE " );
+		if ( entityKeyMetadata.getColumnNames().length == 1 ) {
+			queryBuilder.append( "n." );
+			escapeIdentifier( queryBuilder, entityKeyMetadata.getColumnNames()[0] );
+			queryBuilder.append( " IN {0}" );
+			queryBuilder.append( " RETURN n" );
+		}
+		return queryBuilder.toString();
+	}
+
+	/*
+	 * MATCH (owner:ENTITY:Account {login: {0}}) -[:type]-> (e:EMBEDDED)
+	 * REMOVE e.property
+	 */
+	private Map<String, String> initRemoveEmbeddedPropertyQuery(EntityKeyMetadata entityKeyMetadata, TupleContext tupleContext) {
+		if ( tupleContext == null ) {
+			return Collections.emptyMap();
+		}
+		Map<String, String> removeColumn = new HashMap<>();
+		for ( String column : tupleContext.getSelectableColumns() ) {
+			if ( EmbeddedHelper.isPartOfEmbedded( column ) ) {
+				if ( !removeColumn.containsKey( column ) ) {
+					StringBuilder queryBuilder = new StringBuilder();
+					queryBuilder.append( "MATCH " );
+					appendEntityNode( "n", entityKeyMetadata, queryBuilder );
+					String[] path = EmbeddedHelper.split( column );
+					for ( int i = 0; i < path.length - 1; i++ ) {
+						queryBuilder.append( "-[:" );
+						appendRelationshipType( queryBuilder, path[i] );
+						queryBuilder.append( "]->" );
+						if ( i == path.length - 2 ) {
+							queryBuilder.append( "(e:EMBEDDED) " );
+						}
+						else {
+							queryBuilder.append( "(:EMBEDDED) " );
+						}
+					}
+					queryBuilder.append( "REMOVE e." );
```

Comment:
The problem with it was that it wasn't aligned with the feature in Cypher.
but they might have improve it. I didn't check